mlflow/store/artifact/azure_data_lake_artifact_repo.py

```diff
@@ -0,0 +1,133 @@
+import os
+import posixpath
+import re
+import urllib.parse
+
+
+from mlflow.entities import FileInfo
+from mlflow.exceptions import MlflowException
+from mlflow.store.artifact.artifact_repo import ArtifactRepository
+
+from mlflow.environment_variables import MLFLOW_ARTIFACT_UPLOAD_DOWNLOAD_TIMEOUT
+
+
+def _parse_abfss_uri(uri):
+    """
+    Parse an ABFSS URI in the format
+    "abfss://<file_system>@<account_name>.dbfs.core.windows.net/<path>",
+    returning a tuple consisting of the filesystem, account name, and path
+
+    See more details about ABFSS URIs at
+    https://learn.microsoft.com/en-us/azure/storage/blobs/data-lake-storage-abfs-driver#uri-scheme-to-reference-data
+
+    :param uri: ABFSS URI to parse
+    :return: A tuple containing the name of the filesystem, account name, and path
+    """
+    parsed = urllib.parse.urlparse(uri)
+    if parsed.scheme != "abfss":
```

Comment:
Is `abfs` a valid scheme?

Reply:
It is valid, but is less secure / does not use TLS ([see Azure docs](https://learn.microsoft.com/en-us/azure/storage/blobs/data-lake-storage-introduction-abfs-uri#uri-syntax)), so doesn't seem important to support unless there's a compelling use case